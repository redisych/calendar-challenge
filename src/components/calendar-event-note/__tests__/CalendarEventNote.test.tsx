import * as React from 'react';
import * as enzyme from 'enzyme';
import { CalendarEventNote } from '..';
import toJson from 'enzyme-to-json';

const event = {
    start: 0,
    end: 60,
    widthShift: 0
};

const getWrapper = (): any => {
    return enzyme.shallow(
        <CalendarEventNote
            event={event}
            width={100}
        />
    );
};

describe('CalendarEventNote', () => {
    it('should be defined', () => {
        const wrapper = getWrapper();

        expect(wrapper).toBeDefined();
    });

    it('should match snapshot', () => {
        const wrapper = getWrapper();

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render styles correctly', () => {
        const wrapper = getWrapper();

        expect(wrapper.prop('style')).toHaveProperty('width', 100);
        expect(wrapper.prop('style')).toHaveProperty('height', 60);
        expect(wrapper.prop('style')).toHaveProperty('left', 10);
        expect(wrapper.prop('style')).toHaveProperty('top', 0);
    });

    it('should calculate left offset correctly for widthShift > 0', () => {
        const wrapper = getWrapper();

        const newEvent = {
            ...event,
            widthShift: 1
        };
        wrapper.setProps({
            event: newEvent
        });

        expect(wrapper.prop('style')).toHaveProperty('left', 110);
    });
});
