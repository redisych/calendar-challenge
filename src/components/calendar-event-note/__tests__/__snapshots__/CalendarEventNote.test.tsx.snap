// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarEventNote should match snapshot 1`] = `
<div
  className="calendar-event-note"
  style={
    Object {
      "height": 60,
      "left": 10,
      "top": 0,
      "width": 100,
    }
  }
>
  <div
    className="note-header text-ellipsis"
  >
    Sample Item
  </div>
  <div
    className="note-location text-ellipsis"
  >
    Sample Location
  </div>
</div>
`;
